### ISSUE TITLE
Create a simple backend application that exposes a simple endpoint to register new ingredients into the DB

### ISSUE THREAD

**Issue by felipet** [felipet/lacoctelera_backend#4]
# Description

Add the endpoint `/ingredient` to the backend. The endpoint will support two HTTP functions:

- **GET** to retrieve all the ingredients that match a given string.
- **POST** to submit a new `Ingredient` entry to the DB.

The API shall include documentation to generate an OpenAPI documentation from the source code:
See [utoipa](https://crates.io/crates/utoipa).

## **GET**

The **GET** function of the ingredient receives a query string encoded in the URL to find `Ingredient`s in the DB whose `Ingredient::name` matches partially the query string. The **HTTP** header: `x-www-form-urlencoded` must be included in the request. The response will be encoded as the body of the response using JSON format.

For example:
```bash
$ curl -X GET -H 'Content-Type: application/x-www-form-urlencoded' -i 'http://localhost:8080/ingredient?name=vodka'
```

Shall return:
- If the query is properly built:
  - Return code 200 and a list of ingredients if the DB contains one ingredient at least that matches partially the query string.
  - Return code 200 and an empty list if no existing ingredients matched the query string.
- Otherwise:
  - Return code 400.

## **POST**

The **POST** function of the endpoint receives an `Ingredient` entry (encoded in JSON format) and it handles the addition of the entry into the DB.

For example:
==TODO==

Shall return:
- If the payload is properly built:
  - Return code 200 and a response JSON that contains the ID of the entry in the DB.
  - Return code 200 if there's a key collision in the DB while attempting to add the new `Ingredient` and a JSON that contains the ID of the existing entry in the DB.
- Otherwise:
  - Return code 400.

**Comment by felipet** [comment]
@vera-zhurba the **GET** is ready since c6fd7822ecd292f18613d4bc95c64a79d90176fb . The **POST** is there, but I haven't tested it much, so expected issues while using it. I'll complete the task writing some tests and docs for the **POST**.

The docs use OpenAPI format. If you run the backend, you'd be able to access the docs straight from the root of the API (localhost:9090).

**Comment by felipet** [comment]
The simple implementation of the `/ingredient` endpoint is available since bcc41c1f0a81f9f5c4d0905600d12d4283352daf. Despite it's not polished, it is ready for integration tests with the frontend @vera-zhurba . I'll focus on polishing it now, so I can merge the content into the **main** branch.

**Comment by vera-zhurba** [comment]
Thank you  @felipet! Everything looks great.

**Comment by felipet** [comment]
Merged into devel (356122e)
